laplace/baselaplace.py

```diff
@@ -763,13 +806,22 @@ def posterior_precision(self):
         -------
         precision : `laplace.matrix.KronDecomposed`
         """
-        self._check_fit()
+        if type(self.H) is Kron:  # Fall back to prior
+            return self.prior_precision_diag
         return self.H * self._H_factor + self.prior_precision
 
     @property
     def log_det_posterior_precision(self):
+        if type(self.H) is Kron:  # Fall back to diag prior
+            return self.posterior_precision.log().sum()
```

Comment:
This adds one redundant `if` statement in `posterior_precision`. Could be directly replaced with `prior_precision_diag.log().sum()`.

Reply:
That would still be necessary unless we also change it in `square_norm`. But I will do this because then it's clear that it is falling back to the prior.

Reply:
I see, thanks.